src/components/common/slider-range/slider-range-thumb.vue

```diff
@@ -0,0 +1,92 @@
+<script setup lang="ts">
+import { onUnmounted, Ref, computed, ref } from 'vue'
+
+const props = defineProps<{
+  ariaLabel?: string
+  minValue: number
+  maxValue: number
+  totalSteps: number
```

Comment:
I find it strange to have the steps logic inside the thumb components, I would expect that logic to be done in the parent component, so that the thumb is only given its relative position and will emit an event if moved. Then the parent component will compute if the thumb shall jump to a new position.

I find the current behaviour amazing, it is very clean code, but I found it a bit hard to understand sinci I expected the logics to be in the parent.

Reply:
> I find it strange to have the steps logic inside the thumb components, I would expect that logic to be done in the parent component, so that the thumb is only given its relative position and will emit an event if moved. Then the parent component will compute if the thumb shall jump to a new position.
> 
This is already a dumb component.
The only computation done in this common/slider-range component is for the margin left (for the thumb to be placed in the DOM with the right offset). It emits relative values only (indexes).
The logic for real time values is held by the `layer-time-slider` component.

But maybe the produced code is not very clear or maybe I am missing something?
Which part of the code do you think I should move?